In [51]:
import requests
import pandas as pd

**1.** Make an API request that returns the months where "fire" was searched in 2016. Which month had the most searches?  

In [17]:
endpoint = 'https://data.nashville.gov/resource/fuaa-r5cm.json'
params = {
    'year': '2016',
    'query_text': 'fire'
}
response = requests.get(endpoint, params = params)
response

<Response [200]>

In [31]:
res = response.json()
res

[{'month_name': 'January',
  'year': '2016',
  'query_count': '19',
  'query_text': 'fire'},
 {'month_name': 'February',
  'year': '2016',
  'query_count': '35',
  'query_text': 'fire'},
 {'month_name': 'March',
  'year': '2016',
  'query_count': '32',
  'query_text': 'fire'},
 {'month_name': 'April',
  'year': '2016',
  'query_count': '26',
  'query_text': 'fire'},
 {'month_name': 'May',
  'year': '2016',
  'query_count': '24',
  'query_text': 'fire'},
 {'month_name': 'June',
  'year': '2016',
  'query_count': '31',
  'query_text': 'fire'},
 {'month_name': 'July',
  'year': '2016',
  'query_count': '24',
  'query_text': 'fire'},
 {'month_name': 'August',
  'year': '2016',
  'query_count': '47',
  'query_text': 'fire'},
 {'month_name': 'September',
  'year': '2016',
  'query_count': '36',
  'query_text': 'fire'},
 {'month_name': 'October',
  'year': '2016',
  'query_count': '38',
  'query_text': 'fire'},
 {'month_name': 'November',
  'year': '2016',
  'query_count': '32',
  'query_text

In [97]:
fire_searches = pd.DataFrame({'Month': [res[x].get('month_name') for x in range(0, 12)], 'Searches': [res[x].get('query_count') for x in range(0, 12)]})
fire_searches['Month'] = fire_searches['Month'].astype('string')
fire_searches['Searches'] = fire_searches['Searches'].astype('int')
fire_searches


Month  Searches
0     January        19
1    February        35
2       March        32
3       April        26
4         May        24
5        June        31
6        July        24
7      August        47
8   September        36
9     October        38
10   November        32
11   December        38

In [101]:
total = fire_searches['Searches'].sum()
total

382

In [109]:
fire_searches[fire_searches['Searches']==fire_searches['Searches'].max()]

Month  Searches
7  August        47

**A1:** There were 382 searches for "fire" in 2016, with August being the highest, at 47.

**Q2.** Make an API request that returns all the times a query was run more than 100 times in a month. How many times did this occur?  

In [110]:
endpoint = 'https://data.nashville.gov/resource/fuaa-r5cm.json?$where=query_count > 100.0'
response = requests.get(endpoint)
response

[{'month_name': 'March',
  'year': '2014',
  'query_count': '101',
  'query_text': 'permits'},
 {'month_name': 'January',
  'year': '2015',
  'query_count': '101',
  'query_text': 'criminal court clerk'},
 {'month_name': 'September',
  'year': '2015',
  'query_count': '101',
  'query_text': 'codes'},
 {'month_name': 'March',
  'year': '2016',
  'query_count': '101',
  'query_text': 'police'},
 {'month_name': 'March',
  'year': '2016',
  'query_count': '101',
  'query_text': 'civil service'},
 {'month_name': 'November',
  'year': '2016',
  'query_count': '101',
  'query_text': 'jobs'},
 {'month_name': 'November',
  'year': '2017',
  'query_count': '101',
  'query_text': 'metro holidays'},
 {'month_name': 'November',
  'year': '2017',
  'query_count': '101',
  'query_text': 'longevity pay'},
 {'month_name': 'January',
  'year': '2018',
  'query_count': '101',
  'query_text': 'West Nashville Heights Church of Christ'},
 {'month_name': 'January',
  'year': '2018',
  'query_count': '101',
 

In [112]:
res = response.json()
len(res)

[{'month_name': 'March',
  'year': '2014',
  'query_count': '101',
  'query_text': 'permits'},
 {'month_name': 'January',
  'year': '2015',
  'query_count': '101',
  'query_text': 'criminal court clerk'},
 {'month_name': 'September',
  'year': '2015',
  'query_count': '101',
  'query_text': 'codes'},
 {'month_name': 'March',
  'year': '2016',
  'query_count': '101',
  'query_text': 'police'},
 {'month_name': 'March',
  'year': '2016',
  'query_count': '101',
  'query_text': 'civil service'},
 {'month_name': 'November',
  'year': '2016',
  'query_count': '101',
  'query_text': 'jobs'},
 {'month_name': 'November',
  'year': '2017',
  'query_count': '101',
  'query_text': 'metro holidays'},
 {'month_name': 'November',
  'year': '2017',
  'query_count': '101',
  'query_text': 'longevity pay'},
 {'month_name': 'January',
  'year': '2018',
  'query_count': '101',
  'query_text': 'West Nashville Heights Church of Christ'},
 {'month_name': 'January',
  'year': '2018',
  'query_count': '101',
 

**A2.** There are 1,000 occurances of a query being run more than 100 times in a month.

**Q3.** Make another API request that returns all the times "codes" was searched more than 100 times in a month. How many times did this occur? 

In [115]:
endpoint = 'https://data.nashville.gov/resource/fuaa-r5cm.json?query_text=codes&query'
response = requests.get(endpoint)
response

<Response [200]>

In [116]:
res = response.json()
res

[{'month_name': 'January',
  'year': '2014',
  'query_count': '37',
  'query_text': 'codes'},
 {'month_name': 'February',
  'year': '2014',
  'query_count': '75',
  'query_text': 'codes'},
 {'month_name': 'March',
  'year': '2014',
  'query_count': '90',
  'query_text': 'codes'},
 {'month_name': 'April',
  'year': '2014',
  'query_count': '65',
  'query_text': 'codes'},
 {'month_name': 'May',
  'year': '2014',
  'query_count': '84',
  'query_text': 'codes'},
 {'month_name': 'June',
  'year': '2014',
  'query_count': '77',
  'query_text': 'codes'},
 {'month_name': 'July',
  'year': '2014',
  'query_count': '92',
  'query_text': 'codes'},
 {'month_name': 'August',
  'year': '2014',
  'query_count': '70',
  'query_text': 'codes'},
 {'month_name': 'September',
  'year': '2014',
  'query_count': '71',
  'query_text': 'codes'},
 {'month_name': 'October',
  'year': '2014',
  'query_count': '45',
  'query_text': 'codes'},
 {'month_name': 'November',
  'year': '2014',
  'query_count': '53',
  '